### PR TITLE
UTY-356 Add Steam token config option

### PR DESCRIPTION
**Pull request by Zomtorg**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Implement the option for setting Steam credentials.
#### Tests
Automated unit tests for missing credentials, deployment tag adhering to a regex, and the expected behavior have been implemented.
#### Documentation
None
#### Primary reviewers
@jessicafalk @oblm

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs [hunk 1]
```diff
@@ -16,10 +19,30 @@ public LocatorConfig()
...
+                    ValidateConfig(LocatorParameters.Steam.DeploymentTag, RuntimeConfigNames.SteamDeploymentTag);
+                    ValidateSteamDeploymentTag(LocatorParameters.Steam.DeploymentTag);
+                    break;
+            }
+        }
+
+        private void ValidateSteamDeploymentTag(string deploymentTag)
+        {
+            const string regex = @"^[A-Za-z0-9][A-Za-z0-9_]*$";
+            var match = Regex.Match(deploymentTag, regex);
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs]: I think that it actually needs to be one specific tag:  `steam_deployment`
https://docs.improbable.io/reference/12.0/unitysdk/customize/steam#distributing-with-steam

File: workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs [hunk 2]
```diff
@@ -30,20 +53,48 @@ public void SetProjectName(string projectName)
...
                 parsedArgs, RuntimeConfigNames.LoginToken, string.Empty);
             var projectName = CommandLineUtility.GetCommandLineValue(
                 parsedArgs, RuntimeConfigNames.ProjectName, string.Empty);
+            var steamDeploymentTag = CommandLineUtility.GetCommandLineValue(
+                parsedArgs, RuntimeConfigNames.SteamDeploymentTag, string.Empty);
+            var steamToken = CommandLineUtility.GetCommandLineValue(
+                parsedArgs, RuntimeConfigNames.SteamToken, string.Empty);
+
+            // Assign Steam Credentials first so that loginToken can fall back to default values if not provided
+            config.SetSteamCredentials(steamToken, steamDeploymentTag);
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs]: a login token will fail as well, if it uses the default values. We literally need to provide a login token or steam credentials to go successfully through the locator flow.  So we should have a check here: do we have a steamToken? yes, go set it, no, go set the logintoken

Comment (Zomtorg) [reply workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs]: Having a second look at the implementation I have now baked the logic in the functions to cover the cases you mentioned.

File: workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs (hunk 1 above)
Comment (Zomtorg) [reply workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs]: Here's where I derived the regex from: https://github.com/improbable/everything/blob/3da51c9907e18f0b52172cdb6dfb2c6bef04ce7e/worker_sdk/csharp/sdk/worker/Parameters.cs#L212

We can also hardcode the deployment tag to be "steam_deployment" if that's the case?

**Comment by jamiebrynes7** [comment]
Closing this for the time being - can reopen when it becomes a priority again

**Pull request closed without merging**